workspaces/ui-v2/src/components/sharing/ShareModal.tsx

```diff
@@ -0,0 +1,223 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  Button,
+  makeStyles,
+  TextField,
+  Typography,
+  CircularProgress,
+  IconButton,
+  Tooltip,
+} from '@material-ui/core';
+import CopyIcon from '@material-ui/icons/FileCopy';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useAsyncMemo } from 'use-async-memo';
+import { ShareTarget } from './ShareButton';
+import { useAppSelector } from '<src>/store';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    marginLeft: theme.spacing(3),
+    marginRight: theme.spacing(3),
+  },
+  item: {
+    marginBottom: theme.spacing(2),
+    flex: 1,
+  },
+  logoMarkdownContainer: {
+    overflow: 'scroll',
+    backgroundColor: 'rgba(27, 31, 35, 0.05)',
+    color: 'rgb(36, 41, 46)',
+    padding: theme.spacing(1),
+  },
+}));
+
+function generateBadgeUrl({ endpointCount }: { endpointCount: number }) {
+  const params = {
+    label: 'API Docs',
+    message: `${endpointCount} Endpoints`,
+    style: 'flat',
+    color: 'rgb(43,123,209)',
+    // Took https://github.com/opticdev/optic/blob/develop/website/static/img/logo-bare.svg
+    // Ran it through https://jakearchibald.github.io/svgomg/
+    // base64 encoded it (on macos, you can do: `pbpaste | base64 | pbcopy` to quickly base64 encode whatever is in your paste buffer)
+    // prepend with `data:image/svg+xml;base64,`
```

Comment:
Perhaps this could go in `src/constants`?

Reply:
done